### PR TITLE
feat: dataset generation from data

### PR DESCRIPTION
**Pull request by Toan-Do**
- Added dataset generation from data folder.
- Added example.
- Fixed load HF model bug.
- Disbale batch size finder

### CODE REVIEW

File: src/xturing/self_instruct/prepare_seed_tasks.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    ):
+    text_splitter = RecursiveCharacterTextSplitter(chunk_size=chunk_size)
+
+    texts = text_splitter.split_text(original_text)
+    print(f"Split the document into {len(texts)} parts")
+
+    questions = []
+    answers = []
+    for text in texts:
+        prompt = f"""Given  a document. Suggest {num_samples_per_chunk} questions could be asked related to the document. Generate a comprehensive and informative answer (but no more than 80 words) for each question.
```
Comment (sarthaklangde) [inline src/xturing/self_instruct/prepare_seed_tasks.py:44]:
```suggestion
        prompt = f"""Given  a document. Suggest {num_samples_per_chunk} questions that could be asked related to the document. Generate a comprehensive and informative answer (but no more than 80 words) for each question.
```

**Review by StochasticRomanAgeev — CHANGES_REQUESTED** [review]
Don't we want to use same Engines we were using for instruction dataset generation?

File: examples/dataset_generation/create_instruction_dataset_from_files.ipynb [hunk 2]
```diff
@@ -0,0 +1,109 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Generate dataset from a plain data folder. Some finance news articles are stored in sample_finance_data folder. In this tutorial, we are going to generation Instruction Dataset from a data folder (finance news articles) and perform Instruction fine-tuning on the generated dataset.\n",
```
Comment (sarthaklangde) [inline examples/dataset_generation/create_instruction_dataset_from_files.ipynb:8]:
```suggestion
    "# Generate dataset from a documents folder. 
    
Some finance news articles are stored in `sample_finance_data` folder. In this tutorial, we are going to generation `InstructionDataset` from this folder containing multiple files and then, perform Instruction fine-tuning on the generated dataset.\n",
```

File: examples/dataset_generation/create_instruction_dataset_from_files.ipynb [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# you can specify different model using engine parameter, default is davinci, you also can add strings\n",
+    "# you can provide organization name with organization parameter\n",
+    "engine = Davinci(\"your-api-key\")\n",
+    "dataset = InstructionDataset.generate_dataset_on_data(engine=engine, path=\"./sample_finance_data\")\n",
```
Comment (sarthaklangde) [inline examples/dataset_generation/create_instruction_dataset_from_files.ipynb:40]:
```suggestion
    "dataset = InstructionDataset.generate_dataset_from_dir(engine=engine, path=\"./sample_finance_data\")\n",
```

**Review by sarthaklangde — CHANGES_REQUESTED** [review]

**Review by sarthaklangde — APPROVED** [review]

**Pull request merged**